external-crates/move/crates/move-stackless-bytecode/src/function_target_pipeline.rs

```diff
@@ -288,55 +291,101 @@ impl FunctionTargetsHolder {
+            let is_path_spec: bool =
+                inner_attrs.contains_key_(&AttributeName_::Unknown(Symbol::from("target")));
 
             if !is_verify_spec && !is_focus_spec {
                 self.no_verify_specs.insert(func_env.get_qualified_id());
             }
+
             if is_focus_spec {
                 self.focus_specs.insert(func_env.get_qualified_id());
             } else {
                 self.no_focus_specs.insert(func_env.get_qualified_id());
             }
+
             if inner_attrs.contains_key_(&AttributeName_::Unknown(Symbol::from("ignore_abort"))) {
                 self.ignore_aborts.insert(func_env.get_qualified_id());
             }
 
-            let target_func_env_opt =
-                func_env
-                    .get_name_str()
-                    .strip_suffix("_spec")
-                    .and_then(|name| {
-                        func_env
-                            .module_env
-                            .find_function(func_env.symbol_pool().make(name))
-                    });
-            match target_func_env_opt {
-                Some(target_func_env) => {
-                    self.function_specs.insert(
-                        func_env.get_qualified_id(),
-                        target_func_env.get_qualified_id(),
-                    );
+            if is_path_spec {
+                let function_spec = inner_attrs
+                    .get_(&AttributeName_::Unknown(Symbol::from("target")))
+                    .unwrap();
+
+                if let Attribute_::Assigned(_, boxed_value) = &function_spec.value {
+                    if let AttributeValue_::ModuleAccess(spanned) = &boxed_value.value {
+                        if let ModuleAccess_::ModuleAccess(module_ident, function_name) =
+                            &spanned.value
+                        {
+                            let address = module_ident.value.address;
+                            let module = &module_ident.value.module;
+
+                            let addr_bytes = address.into_addr_bytes();
+                            let module_name = ModuleName::from_address_bytes_and_name(
+                                addr_bytes,
+                                func_env.symbol_pool().make(&module.to_string()),
+                            );
+
+                            if let Some(module_env) =
+                                func_env.module_env.env.find_module(&module_name)
+                            {
+                                let func_sym = func_env.symbol_pool().make(&function_name.value);
+                                if let Some(target_func_env) = module_env.find_function(func_sym) {
+                                    let target_id = target_func_env.get_qualified_id();
+
+                                    if self.function_specs.contains_right(&target_id) {
+                                        panic!("Duplicate target function: {}", function_name.value);
```

Comment:
I see you found my use of `panic` to report errors. That was was a hack, the correct way to report errors is using diagnostics. For example, see #32